src/app/services/filter.service.ts

```diff
@@ -0,0 +1,50 @@
+import {Injectable} from '@angular/core';
+import {Application} from '../shared/application';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FilterService {
+
+  constructor() { }
```

Comment:
Why not use a pipe for these?
https://angular.io/cli/generate#pipe-command
https://angular.io/guide/pipes#pipes